---
title: "TSS analysis"
output: html_document
---

The code below was used for carrying out transcription-start site analysis of the ChIP-Seq/MNase-Seq data.
Prerequisites are:

1. per-base coverage from the sequenceing data (generated by bedtools genomecov)

2. A table of gene locations, downloaded from "annotations" on the organism page at NCBI, saved as a Excel file.

Output is:

1. Average of coverage across all TSS.

2. Heatmap, where each row represents a single TSS. Hierarchical clustering is performed via average linkage.


```{r load_dependencies}
library(readxl)
library(pheatmap)
suppressPackageStartupMessages(library(viridis))
library(ggplot2)
suppressPackageStartupMessages(library(fastcluster))
source('automatedTSSanalysis.R')
source('getTSSdata5.R')
source('getTSSdata6.R')
```

##automatedanalysis:
This section uses the R codes automatedTSSanalysis and getTSSdatato generate a list object which contains 2 things - a matrix containing the sequencing depth across each TSS in the given genome, and a vector made of the column averages of that matrix (the depth across the average of all TSS for that genome). All values in this matrix are normalised to their own mean to allow comparison of different experiments. Further explanation of the dependent codes is within them.

``` {r generate_matrices}
YeastH3TSSoutput=automatedTSSanalysis('H3QIP_perbase.txt',"ProteinTable15_22535.xlsx",2)
HpyATSSoutput=automatedTSSanalysis('hpyA_M4_IP_log_perbase.txt',"ProteinTable1051_300503.xlsx",1)
HcaTrmBTSSoutput=automatedTSSanalysis('Hca_TrmB_Noglu_A.txt',"proteins_1912_171819.xlsx",1)
```

##Plotting heatmaps
Each row represents a single TSS (corresponding to one gene). Each column represents a genomic position near that TSS, with the centre of the heatmap corresponding to the TSS. The colour represents the normalised sequencing depth at that genomic position.
This is a very time and memory-intensive step, especially for the large matrix generated by yeast data. This can be reduced by using only the 800bp nearest the TSS rather than all 1600bp: Yeast$heatmap[,400:1200].

``` {r visualise_heatmaps}
pheatmap(YeastH3TSSoutput$heatmap[,400:1200],cluster_cols=FALSE,color=viridis(10),main="Yeast histone H3 TSS heatmap")
pheatmap(HpyATSSoutput$heatmap,cluster_cols=FALSE,color=viridis(10),main="Hbt sal hpyA TSS heatmap")
```


##Plotting averages
Instead of the heatmap showing data from all genes, plot the average sequencing coverage across all genes of a particular species, enabling easy comparison of how putative histones in different species show different binding characteristics across the TSS.

``` {r avgplots}
combinedTSS=as.data.frame(matrix(nrow=801,ncol=4))
colnames(combinedTSS)=c("Coordinates","Yeast_histone","Hbtsalt_HpyA","Hcahis_TrmB")
combinedTSS$Coordinates=c(-400:400)
combinedTSS$Yeast_histone=YeastH3TSSoutput$meanTSS[400:1200]
combinedTSS$Hbtsal_HpyA=HpyATSSoutput$meanTSS
combinedTSS$Hcahis_TrmB=HcaTrmBTSSoutput$meanTSS
ggplot(combinedTSS, aes(x=Coordinates)) + geom_line(aes(y=Yeast_histone,color = "red"),size=2) +geom_line(aes(y=Hbtsal_HpyA,color="blue"), size=2)+geom_line(aes(y=Hcahis_TrmB,color="palevioletred"), size=1.5)+ylab("Normalised coverage")+xlab("Distance from TSS")+theme(panel.background = element_blank(),panel.grid.major = element_line(size = 0.5, linetype = 'dotted',colour = "black"),panel.border = element_rect(colour = "black", fill=NA,))+ylim(0.5,1.25)+scale_color_identity(labels=colnames(combinedTSS[c(2,4,3)]),guide="legend")+ggtitle("Average binding across TSS - HpyA vs euk. hsitones vs halophilic TFs")
```

Dependency report
```{r session}
sessionInfo()
```
