---
title: "ChIPSeq_visualisation"
author: "Saaz Sakrikar"
date: "1/17/2022"
output: html_document
---

Visualisation of representative ChIP-Seq peaks.

Input is perbase depth data generated by bedtools.

Libraries, chromosome names, and the region to be visualised.
```{r setup,message=FALSE}
library(readxl)
library(bazar)
library(ggplot2)
library(zoo)

Hbt_chr="NC_002607.1"
Hvo_chr="NC_013967.1"

Hbt_regionstart=504000
Hbt_regionstop=514000

Hvo_regionstart=1264000
Hvo_regionstop=1274000
```

Load perbase txt files.
```{r load_data}

#HpyA-HA ChIP-Seq IP and input
HpyA_IP=read.table("hpyA_M3_IP_log_perbase.txt")
HpyA_WCE=read.table("hpyA_M3_WCE_log_perbase.txt")

#HstA-HA ChIP-Seq IP and input
HstA_IP=read.table("hstAIP2_perbase.txt")
HstA_WCE=read.table("hstAWCE2_perbase.txt")
```

Load NCBI protein files (list of genes), restrict to chromosome in question
```{r load_genes}
Hbt_genes=read_excel("ProteinTable1051_300503.xlsx")
Hbt_genes=Hbt_genes[Hbt_genes$`Replicon Accession`==Hbt_chr,]
Hvo_genes=read_excel("proteins_1149_170797.xlsx")
Hvo_genes=Hvo_genes[Hvo_genes$`Replicon Accession`==Hvo_chr,]

```

Normalise IP signal using WCE in the region of interest. Smoothen the curve for visualisation.
```{r normalise}
#HBT

#Only selected chromosome
HpyA_IP=HpyA_IP[HpyA_IP$V1==Hbt_chr,]
HpyA_WCE=HpyA_WCE[HpyA_WCE$V1==Hbt_chr,]

#Get chr-wide IP/WCE average for a baseline
HpyA_IP_avg=mean(HpyA_IP$V3)
HpyA_WCE_avg=mean(HpyA_WCE$V3)
HpyA_baseline=HpyA_IP_avg/HpyA_WCE_avg

#Restrict to selected region
HpyA_IPregion=HpyA_IP[Hbt_regionstart:Hbt_regionstop,]
HpyA_WCEregion=HpyA_WCE[Hbt_regionstart:Hbt_regionstop,]

#Normalise to WCE and smoothen
HpyA_regionnormalised=HpyA_IPregion
HpyA_regionnormalised$V3=HpyA_IPregion$V3/HpyA_WCEregion$V3
HpyA_regionnormalised$V3=rollmean(HpyA_regionnormalised$V3,k=400,fill="extend")


#HstA

#Only selected chromosome
HstA_IP=HstA_IP[HstA_IP$V1==Hvo_chr,]
HstA_WCE=HstA_WCE[HstA_WCE$V1==Hvo_chr,]

#Get chr-wide IP/WCE average for a baseline
HstA_IP_avg=mean(HstA_IP$V3)
HstA_WCE_avg=mean(HstA_WCE$V3)
HstA_baseline=HstA_IP_avg/HstA_WCE_avg

#Restrict to selected region
HstA_IPregion=HstA_IP[Hvo_regionstart:Hvo_regionstop,]
HstA_WCEregion=HstA_WCE[Hvo_regionstart:Hvo_regionstop,]

#Normalise to WCE and smoothen
HstA_regionnormalised=HstA_IPregion
HstA_regionnormalised$V3=HstA_IPregion$V3/HstA_WCEregion$V3
HstA_regionnormalised$V3=rollmean(HstA_regionnormalised$V3,k=400,fill="extend")
```

Converting gene list into visualisable matrix
```{r processgenes}
Hbt_length=length(HpyA_IP$V1)

Hbt_genematrix=as.data.frame(matrix(nrow=Hbt_length,ncol=3))
colnames(Hbt_genematrix)=c("Crd","Lvl","Clr")

Hbt_genematrix$Crd=c(1:Hbt_length)
Hbt_genematrix$Lvl=min(HpyA_regionnormalised$V3-0.1)
Hbt_genematrix$Clr="Intergenic"

Hbt_genelength=length(Hbt_genes$Start)

for (i in (1:Hbt_genelength)) {
  Genestart=Hbt_genes$Start[i]
  Genestop=Hbt_genes$Stop[i]
  if (Hbt_genes$Strand[i]=="-") {
    Hbt_genematrix$Clr[Genestart:Genestop]="Reverse"
    Hbt_genematrix$Lvl[Genestart:Genestop]=min(HpyA_regionnormalised$V3-0.12)
  } else {
    Hbt_genematrix$Clr[Genestart:Genestop]="Forward"
    Hbt_genematrix$Lvl[Genestart:Genestop]=min(HpyA_regionnormalised$V3-0.08)
  }
}

Hbt_genematrix=Hbt_genematrix[Hbt_regionstart:Hbt_regionstop,]


Hvo_length=length(HstA_IP$V1)

Hvo_genematrix=as.data.frame(matrix(nrow=Hvo_length,ncol=3))
colnames(Hvo_genematrix)=c("Crd","Lvl","Clr")

Hvo_genematrix$Crd=c(1:Hvo_length)
Hvo_genematrix$Lvl=min(HstA_regionnormalised$V3-0.08)
Hvo_genematrix$Clr="Intergenic"

Hvo_genelength=length(Hvo_genes$Start)

for (i in (1:Hvo_genelength)) {
  Genestart=Hvo_genes$Start[i]
  Genestop=Hvo_genes$Stop[i]
  if (Hvo_genes$Strand[i]=="-") {
    Hvo_genematrix$Clr[Genestart:Genestop]="Reverse"
    Hvo_genematrix$Lvl[Genestart:Genestop]=min(HstA_regionnormalised$V3-0.1)
  } else {
    Hvo_genematrix$Clr[Genestart:Genestop]="Forward"
    Hvo_genematrix$Lvl[Genestart:Genestop]=min(HstA_regionnormalised$V3-0.06)
  }
}

Hvo_genematrix=Hvo_genematrix[Hvo_regionstart:Hvo_regionstop,]
```

Visualisation and plot export.
```{r plotting}
ggplot() + geom_line(data=HpyA_regionnormalised,mapping=aes(x=V2,y=V3,color=V1),size=1) + theme_bw() + ggtitle("Representative HpyA peak") + xlab("NC_002607.1 genome coordinate") + ylab("IP/Input") + geom_point(data=Hbt_genematrix,mapping=aes(x=Crd,y=Lvl,color=Clr),size=0.5) + scale_color_manual(values=c("darkred","white","darkblue","darksalmon")) + geom_hline(yintercept=HpyA_baseline,colour="gray60",linetype="dashed",size=0.75) + theme(axis.text=element_text(family="sans",size=12,face="bold"),axis.title=element_text(family="sans",size=14,face="bold"))


ggplot() + geom_line(data=HstA_regionnormalised,mapping=aes(x=V2,y=V3,color=V1),size=1) + theme_bw() + ggtitle("Representative HstA peak") + xlab("NC_013967.1 genome coordinate") + ylab("IP/Input") + geom_point(data=Hvo_genematrix,mapping=aes(x=Crd,y=Lvl,color=Clr),size=0.5) + scale_color_manual(values=c("darkred","white","cornflowerblue","darksalmon")) + geom_hline(yintercept=HstA_baseline,colour="gray60",linetype="dashed",size=0.75) + theme(axis.text=element_text(family="sans",size=12,face="bold"),axis.title=element_text(family="sans",size=14,face="bold"))
```


```{r HpyA_plot_export}
# pdf("HpyA_peak_plot.pdf",height=4,width=6)
# 
# ggplot() + geom_line(data=HpyA_regionnormalised,mapping=aes(x=V2,y=V3,color=V1),size=1) + theme_bw() + ggtitle("Representative HpyA peak") + xlab("NC_002607.1 genome coordinate") + ylab("IP/Input") + geom_point(data=Hbt_genematrix,mapping=aes(x=Crd,y=Lvl,color=Clr),size=0.5) + scale_color_manual(values=c("darkred","white","darkblue","darksalmon")) + geom_hline(yintercept=HpyA_baseline,colour="gray60",linetype="dashed",size=0.75) + theme(axis.text=element_text(family="sans",size=12,face="bold"),axis.title=element_text(family="sans",size=14,face="bold"))
# 
# dev.off()
# ```
# ```{r HstA_plot_export}
# pdf("HstA_peak_plot.pdf",height=4,width=6)
# 
# ggplot() + geom_line(data=HstA_regionnormalised,mapping=aes(x=V2,y=V3,color=V1),size=1) + theme_bw() + ggtitle("Representative HstA peak") + xlab("NC_013967.1 genome coordinate") + ylab("IP/Input") + geom_point(data=Hvo_genematrix,mapping=aes(x=Crd,y=Lvl,color=Clr),size=0.5) + scale_color_manual(values=c("darkred","white","cornflowerblue","darksalmon")) + geom_hline(yintercept=HstA_baseline,colour="gray60",linetype="dashed",size=0.75) + theme(axis.text=element_text(family="sans",size=12,face="bold"),axis.title=element_text(family="sans",size=14,face="bold"))
# dev.off()
```

```{r sessioninfo}
sessionInfo()
```

